diff --git a/wgnord b/wgnord
--- a/wgnord
+++ b/wgnord
@@ -24,9 +24,9 @@
 login() {
 	[ -z "$1" ] && help
 	auth_token="$(printf "%s" "token:$1" | base64 -w 0)"
-	query "https://$host/v1/users/services/credentials" -H "Content-Type: application/json" -H "Authorization: Basic $auth_token" > $conf_dir/credentials.json
-	err="$(jq -er '.errors.message' $conf_dir/credentials.json)" && { print_error "Error getting credentials: $err"; exit 1; }
-	echo -n $auth_token > $conf_dir/auth_token
+	credentials=$(query "https://$host/v1/users/services/credentials" -H "Content-Type: application/json" -H "Authorization: Basic $auth_token")
+	err="$(echo "$credentials" | jq -er '.errors.message')" && { print_error "Error getting credentials: $err"; exit 1; }
+	echo "$credentials"
 }
 get_country_code() {
 	cleancode=$(echo "$1" | sed 's/[0-9]*//g')
@@ -35,31 +35,17 @@
 	echo "$buf"
 }
 connect() {
-	[ ! -f $conf_dir/credentials.json ] && {
-		print_error "$conf_dir/credentials.json doesn't exist, are you logged in?"
-		echo -e "\nNote: wgnord changed its login mechanism in version 0.2.0. See --help on how to login."
-		exit 1
-	}
-	out_file=/etc/wireguard/wgnord.conf
-	OPTIND=2
-	while getopts "fno:" o; do case "$o" in
-		f) force=1 ;;
-		n) dont_act=1 ;;
-		o) out_file="$OPTARG" ;;
-		*) exit 1 ;;
-	esac done
+	credentials=$(login "$2") || exit 1
+	template_file="${3:-$conf_dir/template.conf}"
+	force=1
+	dont_act=1
+	out_file="$4"
 	[ -z "$1" ] && help
 	print "Finding best server..."
-	if [ $force ] || [ ! -f $conf_dir/coords.json ] && [ ! $dont_act ]; then
-		is_connected && disconnect
-		insights="$(query "https://$host/v1/helpers/ips/insights")"
-		longitude="$(echo "$insights" | jq -j '.longitude')"
-		latitude="$(echo "$insights" | jq -j '.latitude')"
-		jq -njc "{longitude: $longitude, latitude: $latitude}" > $conf_dir/coords.json
-	else
-		longitude="$(jq -er '.longitude' $conf_dir/coords.json)"
-		latitude="$(jq -er '.latitude' $conf_dir/coords.json)"
-	fi
+	is_connected && disconnect
+	insights="$(query "https://$host/v1/helpers/ips/insights")"
+	longitude="$(echo "$insights" | jq -j '.longitude')"
+	latitude="$(echo "$insights" | jq -j '.latitude')"
 
 	servernr=$(echo "$1" | sed 's/[^0-9]//g')
 	limit=10
@@ -80,8 +66,8 @@
 	server_ip=$(echo "$recommendations" | jq -j --argjson idx "$index" '.[$idx] | .ips[0].ip.ip')
 	server_pubkey=$(echo "$recommendations" | jq -j --argjson idx "$index" '.[$idx] | .technologies[] | select(.metadata[0].name == "public_key") | .metadata[0].value')
 
-	privkey="$(jq -j '.nordlynx_private_key' $conf_dir/credentials.json)"
-	sed -e "s|PRIVKEY|$privkey|" -e "s|SERVER_PUBKEY|$server_pubkey|" -e "s|SERVER_IP|$server_ip|" $conf_dir/template.conf > "$out_file"
+	privkey="$(echo "$credentials" | jq -j '.nordlynx_private_key')"
+	sed -e "s|PRIVKEY|$privkey|" -e "s|SERVER_PUBKEY|$server_pubkey|" -e "s|SERVER_IP|$server_ip|" "$template_file" > "$out_file"
 	if [ ! $dont_act ]; then
 		print "Connecting to $server_hostname ($server_name)..."
 		if is_connected; then
